tests/test_optimizers.py

```diff
@@ -309,33 +311,84 @@ def test_transpose_dequantize_with_axis(self, shape, perm_input, perm_output):
+        axes = list(range(len(input_shape)))
+        for i in range(1, len(input_shape) + 1):
+            axes_combinations.extend(list(itertools.combinations(axes, i)))
+        for axes in axes_combinations:
+            axes = np.array(list(axes), dtype=np.int64)
+            starts = np.array([0] * axes.size, dtype=np.int64)
+            ends = []
+            for i in range(axes.size):
+                ends.append(slice_size[axes[i]])
+            ends = np.array(ends, dtype=np.int64)
+            output_shape = input_shape.copy()
+            for axis in axes:
+                output_shape[perm_input[axis]] = slice_size[axis]
+            node1 = helper.make_node("Transpose", ["X"], ["Y"], perm=perm_input, name="trans_1")
+            node2 = helper.make_node("Slice", ["Y"], ["Z"], starts=starts, ends=ends, axes=axes, name="slice")
+            node3 = helper.make_node("Transpose", ["Z"], ["Z1"], perm=perm_output, name="trans_2")
 
-        graph = helper.make_graph(
-            [node1, node2, node3],
-            "relu-test",
-            [helper.make_tensor_value_info("X", TensorProto.FLOAT, input_shape)],
-            [helper.make_tensor_value_info("Z1", TensorProto.FLOAT, output_shape)],
-            [
-                helper.make_tensor("starts", TensorProto.INT64, starts.shape, starts),
-                helper.make_tensor("ends", TensorProto.INT64, ends.shape, ends),
-                helper.make_tensor("axes", TensorProto.INT64, axes.shape, axes)
-            ]
-        )
+            graph = helper.make_graph(
+                [node1, node2, node3],
+                "slice-test",
+                [helper.make_tensor_value_info("X", TensorProto.FLOAT, input_shape)],
+                [helper.make_tensor_value_info("Z1", TensorProto.FLOAT, output_shape)],
+                [
+                    helper.make_tensor("starts", TensorProto.INT64, starts.shape, starts),
+                    helper.make_tensor("ends", TensorProto.INT64, ends.shape, ends),
+                    helper.make_tensor("axes", TensorProto.INT64, axes.shape, axes)
+                ]
+            )
 
-        model_proto = self.make_model(graph, producer_name="onnx-tests")
-        self.run_transpose_compare(["Z1"], {"X": np.random.randn(*input_shape).astype(np.float32)},
-                                   model_proto, remaining_transpose_num=0)
+            model_proto = self.make_model(graph, producer_name="onnx-tests")
+            self.run_transpose_compare(["Z1"], {"X": np.random.randn(*input_shape).astype(np.float32)},
+                                       model_proto, remaining_transpose_num=0)
+
+    @parameterized.expand([
+        ([2, 3, 4, 5], [1, 2, 1, 2], [0, 2, 3, 1], [0, 3, 1, 2]),
+        ([2, 3, 4, 5, 6], [1, 2, 1, 2, 1], [0, 2, 3, 4, 1], [0, 4, 1, 2, 3]),
+    ])
+    @check_opset_min_version(10, "Slice in opset 10 can accept dynamic 'start' and 'ends'")
+    def test_transpose_slice_10(self, input_shape, slice_size, perm_input, perm_output):
```

Comment:
```suggestion
    def test_transpose_slice_int32(self, input_shape, slice_size, perm_input, perm_output):
```
Nice catch on the type issue.  I know the other tests just have a number but it is generally nice to have a better test description.

Reply:
this test was primarly designed to test Slice from opset 10 (hence the name similarly to test_const_fold_unsqueeze_with_const_13 test case we have). After pretrained_model_test blew up, I decided to change the axes type to int32 in this test case.
Maybe this case should be named like test_transpose_slice_opset_10 - let me know what you think.

Reply:
Ah, my mistake.  Then yes, test_transpose_slice_opset_10 is great.